eng/scripts/automation/Invoke-CadlDataPlaneGenerateSDKPackage.ps1

```diff
@@ -0,0 +1,93 @@
+
+.PARAMETER inputfiles
+The address of the Open API spec files,  separated by semicolon if there is more than one file. 
+The Open API spec file can be local file, or the web address of the file in the `azure-rest-api-specs` repo.
+When pointing to a local file, make sure to use **absolute path**, i.e. /home/swagger/compute.json.
+When pointing to a file in the `azure-rest-api-specs` repo, make sure to include the commit id in the URI, i.e. `https://github.com/Azure/azure-rest-api-specs/blob/73a0fa453a93bdbe8885f87b9e4e9fef4f0452d0/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json`. This ensures that you can choose the time to upgrade to new swagger file versions.
+
+ .PARAMETER readme
+The address of the readme configuration file. The configuration can be local file, e.g. ./swagger/readme.md or the web address of the file in the `azure-rest-api-specs` repo, i.e. `https://github.com/Azure/azure-rest-api-specs/blob/23dc68e5b20a0e49dd3443a4ab177d9f2fcc4c2b/specification/deviceupdate/data-plane/readme.md`
+You need to provide one of `-inputfiles` and `-readme` parameters. If you provide both, `-inputfiles` will be ignored.
+
+.PARAMETER securityScope
+The authentication scope to use if your library supports **Token Credential** authentication.
+
+.PARAMETER securityHeaderName
+The key to use if your library supports **Azure Key Credential** authentication.
+
+.EXAMPLE
+Run script with default parameters.
+
+Invoke-DataPlaneGenerateSDKPackage.ps1 -service <servicename> -namespace Azure.<group>.<service> -sdkPath <sdkrepoRootPath> [-inputfiles <inputfilelink>] [-readme <readmeFilelink>] [-securityScope <securityScope>] [-securityHeaderName <securityHeaderName>]
+
+e.g.
+Invoke-DataPlaneGenerateSDKPackage.ps1 -service webpubsub -namespace Azure.Messaging.WebPubSub -sdkPath /home/azure-sdk-for-net -inputfiles https://github.com/Azure/azure-rest-api-specs/blob/73a0fa453a93bdbe8885f87b9e4e9fef4f0452d0/specification/webpubsub/data-plane/WebPubSub/stable/2021-10-01/webpubsub.json -securityScope https://sample/.default
+
+#>
+param (
+  [string]$service,
+  [string]$namespace,
+  [string]$sdkPath,
+  [string]$cadlRelativeFolder,
+  [string]$specRoot = "",
+  [string]$repo="",
+  [string]$commit="",
+  [string]$additionalSubDirectories="" #additional directories needed, separated by semicolon if more than one
+)
+. (Join-Path $PSScriptRoot GenerateAndBuildLib.ps1)
+$sdkPath = Resolve-Path $sdkPath
+# Generate dataplane library
+$outputJsonFile = "newpackageoutput.json"
+New-CADLPackageFolder `
+  -service $service `
+  -namespace $namespace `
+  -sdkPath $sdkPath `
+  -relatedCadlProjectFolder $cadlRelativeFolder `
+  -specRoot $specRoot `
+  -commit $commit `
+  -repo $repo `
+  -additionalSubDirectories $additionalSubDirectories `
+  -outputJsonFile $outputJsonFile
+
+if ( $? -ne $True) {
+  Write-Error "Failed to create sdk project folder. exit code: $?"
```

Comment:
`$?` is a boolean, so when logging it as an exit code the value will always be false. I'm not sure this conditional will resolve to true though, as the above function would be more likely to throw an exception on failure. Exit codes from external binaries like `dotnet` need to be captured one by one via `$LASTEXITCODE`.

Reply:
I updated to use `$LASTEXITCODE`. Thanks.